modules/core/keeper/msg_server_test.go

```diff
@@ -924,3 +924,140 @@ func (suite *KeeperTestSuite) TestChannelUpgradeTry() {
 		})
 	}
 }
+
+func (suite *KeeperTestSuite) TestChannelUpgradeTimeout() {
+	var (
+		path *ibctesting.Path
+		msg  *channeltypes.MsgChannelUpgradeTimeout
+	)
+
+	cases := []struct {
+		name     string
+		malleate func()
+		expErr   error
+	}{
+		{
+			name:     "success",
+			malleate: func() {},
+			expErr:   nil,
+		},
+		{
+			name: "success with error receipt",
+			malleate: func() {
+				errReceipt := channeltypes.ErrorReceipt{
+					Sequence: 0,
+					Message:  "error",
+				}
+				path.EndpointB.Chain.App.GetIBCKeeper().ChannelKeeper.SetUpgradeErrorReceipt(path.EndpointB.Chain.GetContext(), path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID, errReceipt)
+
+				suite.Require().NoError(path.EndpointB.UpdateClient())
+				suite.Require().NoError(path.EndpointA.UpdateClient())
+
+				upgradeErrorReceiptKey := host.ChannelUpgradeErrorKey(path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID)
+				errorReceiptProof, proofHeight := path.EndpointB.QueryProof(upgradeErrorReceiptKey)
+				counterpartyChannel := path.EndpointB.GetChannel()
+				channelProof, _, _ := path.EndpointA.QueryChannelUpgradeProof()
```

Comment:
maybe add a comment that the channel proof is being regenerated since the proof height will change. The counterparty channel update isn't strictly necessary, but I guess it doesn't hurt